DANS-KNAW/easy-stage-dataset
# Easy 1247

Fixes EASY-1247

#### When applied it will
* Implement the changes as described in the issue.
* Add the helper scripts for the mock application home folder.
* The tests have been cleaned up, so they do not write in the project root anymore.
* A test has been added to check that the `debug-config` stays in synch with the configuration files that are packaged with the finished product.
* Parent pom upgraded to latest version.
* As a consequence: license comments reformatted not to conflict with Scala code formatter anymore. 

#### Where should the reviewer @DANS-KNAW/easy start?
* https://github.com/DANS-KNAW/easy-stage-dataset/compare/master...janvanmansum:EASY-1247?expand=1#diff-0d13331279bc764a20980df7d8dcf552R147
* https://github.com/DANS-KNAW/easy-stage-dataset/compare/master...janvanmansum:EASY-1247?expand=1#diff-99e0c6152cdae0aac2b4f1703bf2f8daR96
* https://github.com/DANS-KNAW/easy-stage-dataset/compare/master...janvanmansum:EASY-1247?expand=1#diff-0d13331279bc764a20980df7d8dcf552R63
* https://github.com/DANS-KNAW/easy-stage-dataset/compare/master...janvanmansum:EASY-1247?expand=1#diff-22807b5cfd0766d5ce087107eaca55ceR101

#### How should this be manually tested?
Do a test run like this.
```
./debug-reset-apphome.sh
./stageDataset.sh -d d -u u -f src/test/resources/external-files-for-medium-ok.txt src/test/resources/deposits/medium data/test-1
```
This will first set up a mock home folder for the application. (A pattern introduced with `easy-bag-store`.) Then the wrapper script will run as if running `easy-stage-dataset`. It will pass the a mapping file that maps two of the files in the input deposit to URL's. You can check this by inspecting:
* `data/test-1/random images_image01_png/cfg.json`, and
*  `data/test-1/a_deeper_path_With some file_txt/cfg.json`
to see that the `EASY_FILE` datastream will be a redirect datastream for these files. The inspect another file item to see that it is still set up to create a managed `EASY_FILE` datastream.

Use the input mapping `external-files-for-medium-fail.txt` to see that the input mapping is checked for non-existent files.

- [x] EASY-1248 must be merged together with this PR.
